internal/provider/template_resource.go

```diff
@@ -1231,8 +1232,9 @@ type LastVersionsByHash = map[string][]PreviousTemplateVersion
 var LastVersionsKey = "last_versions"
 
 type PreviousTemplateVersion struct {
-	ID   uuid.UUID `json:"id"`
-	Name string    `json:"name"`
+	ID     uuid.UUID `json:"id"`
+	Name   string    `json:"name"`
+	TFVars map[string]string
```

Comment:
My understanding is that adding a new field to a struct is backwards compatible - if we read some private state from an older version of the provider the map will just be nil.